tests/components/nest/common.py

```diff
@@ -15,42 +17,87 @@
 
 from tests.common import MockConfigEntry
 
+# Typing helpers
+PlatformSetup = Callable[[], Awaitable[None]]
+T = TypeVar("T")
+YieldFixture = Generator[T, None, None]
+
 PROJECT_ID = "some-project-id"
 CLIENT_ID = "some-client-id"
 CLIENT_SECRET = "some-client-secret"
+SUBSCRIBER_ID = "projects/example/subscriptions/subscriber-id-9876"
 
 CONFIG = {
     "nest": {
         "client_id": CLIENT_ID,
         "client_secret": CLIENT_SECRET,
-        # Required fields for using SDM API
         "project_id": PROJECT_ID,
-        "subscriber_id": "projects/example/subscriptions/subscriber-id-9876",
+        "subscriber_id": SUBSCRIBER_ID,
     },
 }
 
 FAKE_TOKEN = "some-token"
 FAKE_REFRESH_TOKEN = "some-refresh-token"
 
 
-def create_config_entry(hass, token_expiration_time=None) -> MockConfigEntry:
-    """Create a ConfigEntry and add it to Home Assistant."""
+def create_token_entry(token_expiration_time=None):
+    """Create OAuth 'token' data for a ConfigEntry."""
     if token_expiration_time is None:
         token_expiration_time = time.time() + 86400
+    return {
+        "access_token": FAKE_TOKEN,
+        "refresh_token": FAKE_REFRESH_TOKEN,
+        "scope": " ".join(SDM_SCOPES),
+        "token_type": "Bearer",
+        "expires_at": token_expiration_time,
+    }
+
+
+def create_config_entry(token_expiration_time=None) -> MockConfigEntry:
+    """Create a ConfigEntry and add it to Home Assistant."""
     config_entry_data = {
         "sdm": {},  # Indicates new SDM API, not legacy API
         "auth_implementation": "nest",
-        "token": {
-            "access_token": FAKE_TOKEN,
-            "refresh_token": FAKE_REFRESH_TOKEN,
-            "scope": " ".join(SDM_SCOPES),
-            "token_type": "Bearer",
-            "expires_at": token_expiration_time,
-        },
+        "token": create_token_entry(token_expiration_time),
     }
-    config_entry = MockConfigEntry(domain=DOMAIN, data=config_entry_data)
-    config_entry.add_to_hass(hass)
```

Comment:
This is now either (1) inlined in the test or (2) done by the fixture somewhere else. This now seems to copy a pattern I see more commonly in other integration test fixtures where MockConfigEntry is created separately from being added to hass.